great_expectations/zep/context.py

```diff
@@ -131,14 +145,18 @@ def get_context(cls) -> DataContext:
         return cls._context
 
     def __init__(self) -> None:
-        self._sources: _SourceFactories = _SourceFactories()
+        self._sources: _SourceFactories = _SourceFactories(self)
+        self._datasources: Dict[str, Datasource] = {}
         LOGGER.info(f"4a. Available Factories - {self._sources.factories}")
         LOGGER.info(f"4b. `type_lookup` mapping ->\n{pf(self._sources.type_lookup)}")
 
     @property
     def sources(self) -> _SourceFactories:
         return self._sources
 
+    def add_zep_datasource(self, datasource: Datasource) -> None:
+        self._datasources[datasource.name] = datasource
+
```

Comment:
Should this be private? Will we call this outside of the `wrapped` attr call?

Reply:
I've made this private. It's defined on a `DataContext` but is getting called on the `_SourceFactories` class.